src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -135,7 +135,11 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
                 new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
             _state.TcsQueryDataAvailable.Task.ContinueWith((previousTask) => 
                 {
-                    if (previousTask.IsFaulted)
+                    if (token.IsCancellationRequested)
+                    {
+                        _state.TcsReadFromResponseStream.TrySetCanceled();
+                    }
```

Comment:
>If this is necessary to do, it should be done after the subsequent else block that checks for faulted, as if the task failed, its failure should be propagated rather than ignoring the failure and marking it as canceled.
Agreed. If there is a failure already that should be reported first instead of marking the task as canceled. I'll fix that.

>Are we not registering with the CancellationToken elsewhere to trigger cancellation of the operation?
We are not registering the token.  That is a TODO task that probably won't happen until RTM or later.

>If this does need to stay, I'd suggest changing TrySetCanceled() to TrySetCanceled(token).
Until the work with registering the token for cancelling is done (i.e. the TODO), I didn't want to change the design of this a lot considering we're in RC mode.  I will pass in the token to the `TrySetCanceled`.